### DISCUSSION TITLE
Provider/cicruit field information

### DISCUSSION THREAD

**Discussion by deku-m** [netbox-community/netbox#13215] [Ideas]
I want to start the discussion if i can use it as a feature request or not.
As i can understand you might say, you can use custom fields for it. And you get like a table with headline Custom Field with it.
Thats true but i wonder if it is not needed info (required) in the standard overview. 

Is it possible to extend the provider information page?
With things like global phone number/email/support hours/provider-website/maintainance-website?
As the provider now only has ASN and description. 
But many people would like to know this information from the provider i think. And then highlight the primary contact on the first page?

Also the extension of the circuits with IP or IP range the provider is providing for that circuit on that location.
Giving the headsup of which public ip/range is in use.

**Comment by candlerb** [comment]
What Netbox version are you using? Can you describe exactly why custom fields do not work for your use case? What do you mean by "standard overview"?

Custom fields are displayed in detail pages (e.g. `/circuits/providers/1/`), and can be displayed in table views (e.g. `/circuits/providers/` - click the Configure button to control which columns are displayed)

**Reply by deku-m** [reply]
Im using netbox 3.6.1.
Well im not saying they dont work but im suggesting to make it generic for provider information.
As custom field cant be seperated if i want to add different information its all under custom field section and you need to use the grouping tag.

This is with custom fields but i was hoping to make them member of the provider information where ASN/description is positioned.
Example:
![afbeelding](https://github.com/netbox-community/netbox/assets/37069737/79c1a5b4-8738-430a-a193-f45eba404c7e)

Sorry, I dont know what you mean with configure button to control what is displayed as columns.

**Reply by candlerb** [reply]
> As custom field cant be seperated if i want to add different information

I don't know what you mean

> its all under custom field section

Yes

> and you need to use the grouping tag.

You don't need to use grouping. That's optional to control how the custom fields are displayed (their ordering/position/grouping)

> This is with custom fields but i was hoping to make them member of the provider information where ASN/description is positioned

Are you saying your concern is with the screen layout, nothing more?

> Sorry, I dont know what you mean with configure button to control what is displayed as columns.

1. Navigate to Circuits > Providers. This will give you a list of providers. The URL is `/circuits/providers/`
2. At top-right is a button "Configure Table"
    
    ![image](https://github.com/netbox-community/netbox/assets/44789/11b897a9-e151-4e05-86cb-9bbacf3374f3)

3. Click it. You can add and remove columns to the table. All available columns will be shown, including custom fields.

    ![image](https://github.com/netbox-community/netbox/assets/44789/fbbf59b4-31c8-4092-b29f-0946b71d161b)

**Reply by deku-m** [reply]
Well the custom fields works no problem with that. 
Its just a cosmetic thing as the headline is always Custom Fields. When adding different information you will add it under the same custom field table. So the grouping in there will help to seperate this.

My idea was that Provider might have more standard information in general to be added instead of only ASN and description.
My thought was to expand the Provider table with fields like (email, phone,website etc) 
As in the example table I created in the preview marked with the red square.

One other question when using a custom field for email using URL as content.
I do mailto://info@info.com to make it clickable and straight to email app.
Is this expandable in URL so it also support only  when filling in info@info.com it is clickable?

**Reply by candlerb** [reply]
> My thought was to expand the Provider table with fields like (email, phone,website etc)

There is a separate "Contacts" object type for all those fields.

![image](https://github.com/netbox-community/netbox/assets/44789/e4d5b098-fac3-4a61-960f-741452fb11f6)

![image](https://github.com/netbox-community/netbox/assets/44789/9e838e91-5e72-4ca0-8b5c-0b3f642847f1)

![image](https://github.com/netbox-community/netbox/assets/44789/55d89b85-ae92-4d59-b293-562aa540b283)

Apart from that, it sounds to me like you'd prefer than "custom fields" were somehow combined with the "normal" fields in the UI display. I can understand that (and in software anything can be done), but I don't expect it's going to change. The current layout makes it clear what's fixed in core Netbox and what's customizable.

> Is this expandable in URL so it also support only when filling in [info@info.com](mailto:info@info.com) it is clickable?

No, because without the `mailto:` prefix it's not a URL, and there's no custom field type specifically for "email".
